#! /bin/sh
patch -p1 -l -f $* < $0
exit $?

From 19a4c4daa2cfd075b181d5131a4b8b4d54714b8c Mon Sep 17 00:00:00 2001
From: Andre Heinecke <aheinecke@gnupg.org>
Date: Wed, 27 Mar 2019 16:12:38 +0100
Subject: [PATCH 1/2] core: Fix assuan logger-fd hack for windows

* src/assuan-support.c (my_spawn): Zero is a perfectly fine fd.
---
 src/assuan-support.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/assuan-support.c b/src/assuan-support.c
index 925aebaf..09db3295 100644
--- a/src/assuan-support.c
+++ b/src/assuan-support.c
@@ -234,7 +234,7 @@ my_spawn (assuan_context_t ctx, pid_t *r_pid, const char *name,
                   break;
                 }
               logger_fd = strtol (argv[loc], &tail, 10);
-              if (tail == argv[loc] || logger_fd <= 0)
+              if (tail == argv[loc] || logger_fd < 0)
                 {
                   err = GPG_ERR_INV_ARG;
                   break;
--
2.20.1
